applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -178,12 +270,23 @@ static void handle_dfu_start(const struct config_event *event)
 			LOG_INF("Restart DFU");
 		}
 	} else {
-		img_length = length;
-		img_csum = csum;
-		cur_offset = 0;
+		if (cur_offset != 0) {
+			k_delayed_work_submit(&background_erase, 0);
+			is_flash_area_clean = false;
+			cur_offset = 0;
```

Comment:
will the same call work correctly when called again just after this one?

Reply:
When `handle_dfu_start` is called twice in a row, the first call will submit the work and second one will notice that `is_flash_area_clean` is not `true` and return before reaching this code.

Reply:
Only warning will be logged: "Flash is not clean yet."